From ee25425f987eeddbe6b9fb270089239fe985e9b5 Mon Sep 17 00:00:00 2001
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Mon, 19 Oct 2015 13:33:10 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: When multiple planes are enabled
 configure pfit to AUTO mode

Per BSPEC if pfit has to be enabled in Pillarbox or
Letterbox mode destination should have wider or taller
aspect ratio than the source. But with multiple planes
enabled each plane with different aspect ratio panel fitter
can be enabled in AUTO mode.

This patch tobe squashed with the below patch
commit 4804b54d8cdc0ce2a0a6b31ca49b533b3b30924b
Author: Arun R Murthy <arun.r.murthy@intel.com>
Date:   Tue Aug 11 14:29:54 2015 +0530

    SQUASHME! [VPG]: calculate the aspect ratio for pfit

Change-Id: I81ca802db196556fdbcaa3af218ca181859b4c01
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7475
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index b6edfb0..72aea74 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11235,6 +11235,16 @@ int intel_set_disp_calc_flip(struct drm_mode_set_display *disp,
 					(disp->panel_fitter.src_w / disp->panel_fitter.src_h))
 				pfit_mode |= PFIT_SCALING_AUTO;
 
+			/*
+			 * After calculating the aspect ratio and determining the mode
+			 * check for multi-planes. If multi planes panel fitter has to
+			 * be enabled only in AUTO mode.
+			 */
+			if (!dev_priv->maxfifo_enabled) {
+				pfit_mode &= MASK_PFIT_SCALING_MODE;
+				pfit_mode |= PFIT_SCALING_AUTO;
+			}
+
 			/* Enable Panel fitter if any valid mode is set */
 			pfit_control = (1 << 31) | pfit_control;
 			if (disp->panel_fitter.mode == AUTOSCALE) {
-- 
1.9.1

